runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -1959,6 +1982,24 @@ TR_J9InlinerPolicy::inlineUnsafeCall(TR::ResolvedMethodSymbol *calleeSymbol, TR:
       case TR::sun_misc_Unsafe_putObjectOrdered_jlObjectJjlObject_V:
          return createUnsafePutWithOffset(calleeSymbol, callerSymbol, callNodeTreeTop, callNode, TR::Address, false, true, true);
 
+      // FIXME: Update createUnsafePutWithOffset signature to have isVolatile, isOrdered, isUnaligned as enum
+      case TR::jdk_internal_misc_Unsafe_getCharUnaligned:
+         return createUnsafeGetWithOffset(calleeSymbol, callerSymbol, callNodeTreeTop, callNode, TR::Int8, /*needsNullCheck*/false, /*isUnaligned*/true);
```

Comment:
For `char` the type should be `Int16`. Similarly for `putCharUnaligned`

Reply:
Also, for integral types narrower than 32-bit, the type we pass in doesn't distinguish between signed and unsigned, which matters when we load because the result needs to be extended. When `createUnsafeGetWithOffset()` generates the extension, it switches on the recognized method again to decide between sign-extension and zero-extension. That switch will need to handle `getCharUnaligned()`